kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java

```diff
@@ -120,7 +127,7 @@ public void open(Collection<TopicPartition> partitions) {
     assignment.addAll(partitions);
     for (TopicPartition tp : assignment) {
       TopicPartitionWriter writer =
-          new TopicPartitionWriter(tp, storage, writerProvider, partitioner, connectorConfig, context);
+          new TopicPartitionWriter(tp, writerProvider, partitioner, connectorConfig, context, time);
```

Comment:
There's a comment above this `TopicPartitionWriter` constructor that says `visible for testing`. That's no longer true and should be deleted.

There's also some cleanup that seems possible since now the other `TopicPartitionWriter` constructor is only used for testing, has an unused parameter, and when removed could just switch all the tests to using the one used here (they'd just have to pass in `Time.SYSTEM` as one final parameter). But we can always clean that up later.